src/utils.js

```diff
@@ -34,3 +34,33 @@ module.exports.reflect = async function (thenable, args) {
     return error;
   }
 };
+
+/**
+ * Creates promise that will resolve after the given time (in ms).
+ */
+
+module.exports.timeout = function (ms, returnValue) {
+  return new Promise(resolve => setTimeout(() => resolve(returnValue), ms));
+};
+
+/**
+ * Validate logger.
+ */
+
+module.exports.validateLogger = function (logger) {
+  if (typeof logger !== 'object') {
+    throw new Error(`Logger instance is invalid`);
+  }
+  const requiredLogMethods = ['info', 'warn', 'error'];
+
+  for (const logMethod of requiredLogMethods) {
+    if (!logger?.[logMethod]) {
```

Comment:
```suggestion
    if (!logger.[logMethod]) {
```

Provided the condition above, would the logger ever reach this step as falsy?
```sh
  if (typeof logger !== 'object') {
     throw new Error(`Logger instance is invalid`);
   }
```